department-of-veterans-affairs/caseflow
# MattT/Remove :vha_irregular_appeals FeatureToggle

## Description

Remove usage of the `:vha_irregular_appeals` FeatureToggle as it is no longer needed. The options it was used to restrict on the CompleteTaskModal.jsx component are being opened up to every user.

## Acceptance Criteria
- [ ] Whenever a VHA CAMO performs a "Send to Board Intake" action on an appeal that is in their queue, and the `vha_irregular_appeals` FeatureToggle is disabled, the following radio field options appear under "What is the status of this appeal?" on the CompleteTaskModal:
   - [ ] This issue might not be appealable to the Board
   - [ ] A VHA decision has not been made
   - [ ] Appeal is not related to a VHA issue
- [ ] This radio field option that was not locked behind the FeatureToggle still appears: "Correct documents have been successfully added"
- [ ] All usage of `vha_irregular_appeals` has been stripped from the code without introducing any regression of functionality.

## Testing Plan
1. Open the Rails console, and disable the `vha_irregular_appeals` FeatureToggle:
```ruby
pry(main)> FeatureToggle.disable!(:vha_irregular_appeals)
```

2. Log in as an Intake user (ex: `BVAISHAW`) and perform a 10182 Intake. Whenever adding an issue to the appeal, select the `Veterans Health Administration` benefit type. The issue category that you select does not matter.

3. After submitting the appeal, click on "correct the issues" on the completion page. Grab the appeal's UUID from the URL.

4. Switch to a VHA CAMO user (ex: `CAMOADMIN`)

5. Navigate to the Case Details page for the new appeal:
`/queue/appeals/<APPEAL-UUID>`

6. Click on the Actions dropdown, and select "Send to Board Intake".

7. Ensure that all four radio field options are present:
   1. Correct documents have been successfully added
   2. This issue might not be appealable to the Board
   3. A VHA decision has not been made
   4. Appeal is not related to a VHA issue

8. Make sure you are able to select any one of these options and click the "Submit" button. The `VhaDocumentSearchTask` on the appeal should be completed:
```ruby
# Save appeal to variable for subsequent usage
a = Appeal.find_by_uuid("<APPEAL-UUID>")

# Prints out the appeal's task tree with formatting to make it easier to read
a.treee
```

9. To be able to test the other radio field options, you can create a new `VhaDocumentSearchTask` on the appeal and then refresh the case details page to have another go:

You will need the id of the `PreDocketTask` on the appeal that is obtained from running `a.treee`.
```ruby
    VhaDocumentSearchTask.create!(appeal: a, assigned_by: User.find_by_css_id("BVAISHAW"), assigned_to: VhaCamo.singleton, parent: Task.find(<ID-of-PreDocketTask>)
```

## User Facing Changes

The before and after screenshots were taken with the `vha_irregular_appeals` FeatureToggle disabled.

| Before | After |
| --- | --- |
![localhost_3000_queue_appeals_f0dda0f9-cb6a-4da6-916a-675a2adec463_tasks_7086_modal_vha_send_to_board_intake](https://user-images.githubusercontent.com/99351305/180212918-323cea8d-c7e7-4a68-b071-528ad59a6769.png)|![localhost_3000_queue_appeals_f0dda0f9-cb6a-4da6-916a-675a2adec463](https://user-images.githubusercontent.com/99351305/180212954-d4860672-48e4-45c6-bcbd-23da9246d81f.png)|
